app/src/store/features/rulesActions.ts

```diff
@@ -151,17 +150,28 @@ export const updateRulePairAtGivenPath = (
 ) => {
   const { pairIndex, updates = {}, triggerUnsavedChangesIndication = true } = action.payload;
 
+  let updated = false;
   for (const [modificationPath, value] of Object.entries(updates)) {
-    set(
-      (prevState.rules.currentlySelectedRule.data as Rule)?.pairs?.[pairIndex],
-      getFilterObjectPath(modificationPath),
-      value
-    );
+    if (isRuleModificationAllowed(prevState.rules.currentlySelectedRule.data as Rule, modificationPath)) {
+      set(
+        (prevState.rules.currentlySelectedRule.data as Rule)?.pairs?.[pairIndex],
+        getFilterObjectPath(modificationPath),
+        value
+      );
+      updated = true;
+    }
   }
 
-  if (triggerUnsavedChangesIndication) {
+  if (updated && triggerUnsavedChangesIndication) {
```

Comment:
Instead of maintaining the `updated` flag, can't we directly use `isRuleModificationAllowed` ?

Reply:
Since this is a very core function I was trying to keep it flexible. 

updates do have multiple entries which, in the future, could only partially be allowed. 
Hence wrote it this way.

unsaved changes need to be triggered even when partial updates are allowed.

P.S. Just anticipating that there could be further checks/fixes here in the future because of the re-rendering issue. But yes, certainly could be made simpler right now. wdyt?

Reply:
the `updated` flag is bit hard to read and understand. Its usage is not clear just by looking at the code. `isRuleModificationAllowed` conveys the usage better